Training/Program.cs

```diff
@@ -1,7 +1,133 @@
+         if (Count >= Capacity) Array.Resize (ref mVariables, Capacity * 2);
+         mVariables[mSize++] = element;
+      }
+
+      /// <summary>Removes the first occurrence of a specific element from the list</summary>
+      /// <param name="element">The element to remove</param>
+      /// <returns>True if the element was successfully removed</returns>
+      public bool Remove (T element) {
+         int index = Array.IndexOf (mVariables, element, 0, Count);
+         if (index >= 0) {
+            RemoveAt (index);
+            return true;
+         }
+         return false;
+      }
+
+      /// <summary>Clears the entire list</summary>
+      public void Clear () => mSize = 0;
+
+      /// <summary>Insert elements at a specified index</summary>
+      /// <param name="index"></param>
+      /// <param name="element"></param>
+      /// <exception cref="IndexOutOfRangeException"></exception>
+      public void Insert (int index, T element) {
+         if (Count >= Capacity) Array.Resize (ref mVariables, Capacity * 2);
+         for (int i = mSize; i > index; i--) mVariables[i] = mVariables[i - 1];
+         mVariables[index] = element;
+         mSize++;
+      }
+
+      /// <summary>Removes the element at the specified index</summary>
+      /// <param name="index"></param>
+      /// <exception cref="IndexOutOfRangeException"></exception>
+      public void RemoveAt (int index) {
+         if (index < 0 || index >= mVariables.Length) throw new IndexOutOfRangeException ("Index is out of range");
+         for (int i = index; i < mVariables.Length - 1; i++)
+            mVariables[i] = mVariables[i + 1];
+         Array.Resize (ref mVariables, mVariables.Length - 1);
+         mSize--;
+      }
+      #endregion
+
+      #region Private Fields ------------------------------------------------
+      T[] mVariables;
+      int mSize = 0;
+      #endregion
+   }
+   #endregion
+
+   /// <summary>The Main method of the MyList class is demonstrated</summary>
+   internal class Program {
+      #region Methods -----------------------------------------------
+      /// <summary>Entry Point of the program</summary>
+      static void Main (string[] args) {
+         MyList<int> list = new ();
+         Console.WriteLine ($"Count before adding elements in the list: {list.Count}");
+         Console.WriteLine ($"Capacity before adding elements: {list.Capacity}");
+         list.Add (0);
+         list.Add (1);
+         list.Add (2);
+         list.Add (4);
+         list.Remove (0);
+         list.Add (5);
+         list.Add (6);
+         list.Add (7);
+         PrintMyList (list);
+         Console.WriteLine ($"Count after adding elements in the list: {list.Count}");
+         Console.WriteLine ($"Capacity after adding elements: {list.Capacity}");
+         list.RemoveAt (0);
+         list.Insert (1, 3);
+         PrintMyList (list);
+         list.Clear ();
+         Console.Write ("After clearing list, list count => ");

```

Comment:
Can write it in a single line   Console.Write ($"After clearing list, list count => {list.Count}");